Makefile

```diff
@@ -50,56 +65,45 @@ record_%: FORCE
 clean_%: FORCE
 	ansible-playbook --tags teardown,cleanup -i tests/inventory/hosts 'tests/test_playbooks/$*.yml'
 
-sanity: dist-install
-	ansible-playbook $(CURDIR)/tests/extras/sanity.yml -e test_collection_path=$(COLLECTION_INSTALL_TMP)/ansible_collections/theforeman/foreman/
-
-debug:
-ifndef MODULE
-	$(error MODULE is undefined)
-endif
-	./.tmp/ansible/hacking/test-module $(DEBUG_OPTIONS)
-
-setup: test-setup debug-setup
-
-debug-setup: .tmp/ansible
-.tmp/ansible:
-	ansible-playbook debug-setup.yml
+setup: test-setup
 
 test-setup: tests/test_playbooks/vars/server.yml
 	pip install --upgrade 'pip<20'
 	pip install -r requirements-dev.txt
-	pip install -r https://raw.githubusercontent.com/ansible/ansible/devel/requirements.txt
+
 tests/test_playbooks/vars/server.yml:
 	cp tests/test_playbooks/vars/server.yml.example tests/test_playbooks/vars/server.yml
+	@echo "For recording, please adjust tests/tests_playbooks/vars/server.yml to match your reference server."
 
-dist:
-	mkdir -p $(COLLECTION_TMP)
-
-	# only copy selected files/folders from our git
-	git archive HEAD LICENSE README.md galaxy.yml plugins tests/sanity | tar -C $(COLLECTION_TMP) -xf -
+dist-test: $(MANIFEST)
+	ANSIBLE_COLLECTIONS_PATHS=build/collections ansible -m theforeman.foreman.foreman_organization -a "username=admin password=changeme server_url=https://foreman.example.test name=collectiontest" localhost | grep -q "Failed to connect to Foreman server"
+	ANSIBLE_COLLECTIONS_PATHS=build/collections ansible-doc theforeman.foreman.foreman_organization | grep -q "Manage Foreman Organization"
 
-	# fix the imports to use the collection namespace
-	sed -i '/ansible.module_utils.foreman_helper/ s/ansible.module_utils/ansible_collections.theforeman.foreman.plugins.module_utils/g' $(COLLECTION_TMP)/plugins/modules/*.py
-	sed -i -e '/extends_documentation_fragment/{:1 n; s/- foreman/- theforeman.foreman.foreman/; t1}' $(COLLECTION_TMP)/plugins/modules/*.py
+$(MANIFEST): $(NAMESPACE)-$(NAME)-$(VERSION).tar.gz
+	ansible-galaxy collection install -p build/collections $< --force
 
-	# adjust README.md not to point to files that we don't ship in the collection
-	sed -i '/Documentation how to/d' $(COLLECTION_TMP)/README.md
+# fix the imports to use the collection namespace
+build/src/plugins/modules/%.py: plugins/modules/%.py | build
```

Comment:
what's `| build`?

Reply:
Build is a directoy, whose mtime will always be that of the last file changed in it.
So by declearing it a 'order-only-prerequisite' we can assure it is only called if the directory does not yet exist.

Reply:
🤯